From 8b7b46e31cce51c91f1399630455b49125d144b3 Mon Sep 17 00:00:00 2001
From: Kevin O'Connor <kevin@koconnor.net>
Date: Mon, 28 Sep 2026 21:42:36 -0400
Subject: [PATCH 3/6] ar100: Utilize timer_dispatch_many_polling()

Convert to timer_dispatch_many_polling() and check for serial activity
in the dispatch loop.

Signed-off-by: Kevin O'Connor <kevin@koconnor.net>
(cherry picked from commit 518a452ef41081bc6391054377c240339c769221)
---
 src/ar100/main.c | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/src/ar100/main.c b/src/ar100/main.c
index 791b58a9c..aa391e1f9 100644
--- a/src/ar100/main.c
+++ b/src/ar100/main.c
@@ -64,6 +64,19 @@ void
 irq_wait(void)
 {
     irq_poll();
+    timer_dispatch_task_poll();
+}
+
+void
+timer_dispatch_irq_poll(void)
+{
+}
+
+void
+timer_dispatch_task_poll(void)
+{
+    if(r_uart_fifo_rcv())
+        sched_wake_task(&console_wake);
 }
 
 void
@@ -71,11 +84,9 @@ irq_poll(void)
 {
     if(timer_interrupt_pending()) {
         timer_clear_interrupt();
-        uint32_t next = timer_dispatch_many();
+        uint32_t next = timer_dispatch_many_polling();
         timer_set(next);
     }
-    if(r_uart_fifo_rcv())
-        sched_wake_task(&console_wake);
 }
 
 /****************************************************************
-- 
2.47.3

